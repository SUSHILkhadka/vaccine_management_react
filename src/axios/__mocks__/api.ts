import { vaccineArray } from '../../__test__/utils/sort.test';

module.exports = {
  post: (url: string, body: any = "without any body") => {
    return {
      data: {
        data: body,
      },
    };
  },
  get: (url: string) => {
    return {
      data: {
        data: vaccineArray,
      },
    };
  },
  put: (url: string, body: any) => {
    return {
      data: {
        data: body,
      },
    };
  },
  delete: (url: string, body: any = "without any body") => {
    return {
      data: {
        data: body,
      },
    };
  },
};
