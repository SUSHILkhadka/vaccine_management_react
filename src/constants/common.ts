// export const URL_TO_BACKEND = 'http://localhost:8000';
export const URL_TO_BACKEND = 'https://fierce-peak-73671.herokuapp.com';
// export const imageURL = 'https://www.gravatar.com/avatar/2c7d99fe281ecd3bcd65ab915bac6dd5?s=250';
export const imageURL =
  'https://images.theconversation.com/files/411706/original/file-20210716-21-lunh2y.jpg?ixlib=rb-1.1.0&q=45&auto=format&w=1200&h=1200.0&fit=crop';
export const IMAGE_COMPRESSION_PERCENTAGE = 100;
export const FIXED_FILE_SIZE = 200 * 1000; //around 200kb
export const IMAGE_LOGO1 = '../../assets/logo1.png';
export const IMAGE_LOGO2 = '../../assets/logo2.png';
