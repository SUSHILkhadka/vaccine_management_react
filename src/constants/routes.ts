export const PATH_ROOT = '/';
export const PATH_LOGIN = '/login';
export const PATH_REGISTER = '/register';
export const PATH_SETTING = '/setting';
export const PATH_VACCINE_ADD = '/vaccine/add';
export const PATH_VACCINE_TABLE = '/';
export const PATH_VACCINE_EDIT = '/vaccine/edit';
