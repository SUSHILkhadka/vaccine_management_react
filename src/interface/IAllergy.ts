export type IAllergy = {
    id: number;
    name: string;
    status: 'read' | 'added' | 'edited' | 'deleted';
  };
  