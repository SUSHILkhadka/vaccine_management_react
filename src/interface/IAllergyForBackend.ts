export interface IAllergyForBackend {
  id: number;
  name: string;
  vaccineId: number;
}
export type IAllergyForBackendToInsert = Omit<IAllergyForBackend, 'id'>;
