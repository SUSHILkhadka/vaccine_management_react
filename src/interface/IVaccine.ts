export interface IVaccine {
  id: number;
  name: string;
  description: string;
  numberOfDoses: number;
  releaseDate: string;
  photoUrl: string;
  isMandatory: boolean;
}

export type IVaccineToInsert = Omit<IVaccine, 'id'>;
