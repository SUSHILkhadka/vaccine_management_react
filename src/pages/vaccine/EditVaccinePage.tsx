import { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { getAllAllergiesByPatientId } from '../../axios/backendAllergy';
import VaccineForm from '../../components/vaccine/VaccineForm';
import {
  loadAllergyList,
  resetAllergyList,
} from '../../redux_toolkit/slices/allergySlice';
import { RootState } from '../../redux_toolkit/stores/store';
import '../../styles/Page.scss';
import { showDefaultErrorMessage } from '../../utils/message';
const EditVaccinePage = () => {
  const vaccineInfo = useSelector((state: RootState) => state.vaccine);
  const dispatch = useDispatch();

  useEffect(() => {
    let isMounted = true;
    dispatch(resetAllergyList());

    const getAllAllergyOfVaccine = async () => {
      try {
        const response = await getAllAllergiesByPatientId(vaccineInfo.id);
        if (isMounted) {
          dispatch(loadAllergyList(response.data));
        }
      } catch {
        showDefaultErrorMessage();
      }
    };
    getAllAllergyOfVaccine();

    return () => {
      isMounted = false;
    };
  }, []);

  const initialValue = {
    id: vaccineInfo.id,
    name: vaccineInfo.name,
    description: vaccineInfo.description,
    numberOfDoses: vaccineInfo.numberOfDoses,
    releaseDate: vaccineInfo.releaseDate,
    photoUrl: vaccineInfo.photoUrl,
    isMandatory: vaccineInfo.isMandatory,
  };

  return (
    <div className='page--addvaccine'>
      Edit page
      <VaccineForm initialValue={initialValue} />
    </div>
  );
};

export default EditVaccinePage;
