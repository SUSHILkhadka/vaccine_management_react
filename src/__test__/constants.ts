import { IAllergy } from '../interface/IAllergy';
import { IVaccine } from '../interface/IVaccine';

export const vaccine: IVaccine = {
  id: 24,
  name: 'FluMist',
  description: 'Used in Seasonal Flu. Its other name is LAIV.',
  numberOfDoses: 2,
  releaseDate: '2008-01-30',
  photoUrl: '',
  isMandatory: false,
};

export const allergyList: IAllergy[] = [
  {
    id: -1,
    name: 'allery1',
    status: 'added',
  },
  {
    id: 3,
    name: 'allery3',
    status: 'added',
  },
  {
    id: 3,
    name: 'allery3',
    status: 'deleted',
  },
];

export const vaccineArray = [
  {
    id: 1,
    name: 'a',
    description: '',
    numberOfDoses: 1,
    releaseDate: new Date().toLocaleDateString(),
    photoUrl: '',
    isMandatory: false,
  },
  {
    id: 2,
    name: 'vaccinenameiswired',
    description: '',
    numberOfDoses: 1,
    releaseDate: new Date().toLocaleDateString(),
    photoUrl: '',
    isMandatory: true,
  },
  {
    id: 3,
    name: 'a',
    description: '',
    numberOfDoses: 1,
    releaseDate: new Date().toLocaleDateString(),
    photoUrl: '',
    isMandatory: false,
  },
  {
    id: 4,
    name: 'a',
    description: '',
    numberOfDoses: 1,
    releaseDate: new Date().toLocaleDateString(),
    photoUrl: '',
    isMandatory: true,
  },
];
