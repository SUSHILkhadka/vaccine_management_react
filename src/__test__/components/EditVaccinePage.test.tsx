import { IVaccine } from '../../interface/IVaccine';
import EditVaccinePage from '../../pages/vaccine/EditVaccinePage';
import '../../__mocks__/matchMedia';
import { allergyList, vaccine } from '../constants';
import { renderWithProvidersAndNavigation } from './customRender';


it('should render edit vaccine page properly', () => {
  const page = renderWithProvidersAndNavigation(<EditVaccinePage />, {
    preloadedState: {
      vaccine: vaccine,
      allergy: allergyList,
    },
  });

  expect(page.asFragment()).toMatchSnapshot();
});
