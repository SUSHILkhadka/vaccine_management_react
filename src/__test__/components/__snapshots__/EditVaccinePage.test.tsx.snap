// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render edit vaccine page properly 1`] = `
<DocumentFragment>
  <div
    class="page--addvaccine"
  >
    <div>
      <form
        autocomplete="off"
        class="ant-form ant-form-vertical ant-form-hide-required-mark"
      >
        <div
          class="form--vaccine--header--container"
        >
          <div
            class="ant-row"
          >
            <div
              class="ant-col ant-col-12 form--vaccine--header--left"
            >
              <span
                class=""
              >
                <div
                  class="ant-upload ant-upload-select ant-upload-select-text"
                >
                  <span
                    class="ant-upload"
                    role="button"
                    tabindex="0"
                  >
                    <input
                      accept=""
                      style="display: none;"
                      type="file"
                    />
                    <div
                      class="customphoto--container"
                    >
                      <div
                        class="image--container"
                      >
                        <img
                          alt="loading"
                          class="img--avatar"
                          src="https://images.theconversation.com/files/411706/original/file-20210716-21-lunh2y.jpg?ixlib=rb-1.1.0&q=45&auto=format&w=1200&h=1200.0&fit=crop"
                        />
                      </div>
                      <div
                        class="icon--container"
                      >
                        <div
                          class="icon--plus"
                        >
                          +
                        </div>
                      </div>
                    </div>
                  </span>
                </div>
              </span>
            </div>
            <div
              class="ant-col ant-col-12 form--vaccine--header--right"
            >
              <button
                class="ant-btn ant-btn-default button button--save"
                type="submit"
              >
                <span>
                  Save
                </span>
              </button>
            </div>
          </div>
        </div>
        <div
          class="form--row2"
        >
          <div
            class="form form--vaccine"
          >
            <div
              class="ant-form-item"
            >
              <div
                class="ant-row ant-form-item-row"
              >
                <div
                  class="ant-col ant-col-24 ant-col-offset-1 ant-form-item-label"
                >
                  <label
                    class=""
                    for="name"
                    title="Vaccine name"
                  >
                    Vaccine name
                  </label>
                </div>
                <div
                  class="ant-col ant-col-24 ant-form-item-control"
                >
                  <div
                    class="ant-form-item-control-input"
                  >
                    <div
                      class="ant-form-item-control-input-content"
                    >
                      <span
                        class="ant-input-affix-wrapper input ant-input-affix-wrapper-has-feedback"
                      >
                        <span
                          class="ant-input-prefix"
                        >
                          <span
                            aria-label="user"
                            class="anticon anticon-user"
                            role="img"
                          >
                            <svg
                              aria-hidden="true"
                              data-icon="user"
                              fill="currentColor"
                              focusable="false"
                              height="1em"
                              viewBox="64 64 896 896"
                              width="1em"
                            >
                              <path
                                d="M858.5 763.6a374 374 0 00-80.6-119.5 375.63 375.63 0 00-119.5-80.6c-.4-.2-.8-.3-1.2-.5C719.5 518 760 444.7 760 362c0-137-111-248-248-248S264 225 264 362c0 82.7 40.5 156 102.8 201.1-.4.2-.8.3-1.2.5-44.8 18.9-85 46-119.5 80.6a375.63 375.63 0 00-80.6 119.5A371.7 371.7 0 00136 901.8a8 8 0 008 8.2h60c4.4 0 7.9-3.5 8-7.8 2-77.2 33-149.5 87.8-204.3 56.7-56.7 132-87.9 212.2-87.9s155.5 31.2 212.2 87.9C779 752.7 810 825 812 902.2c.1 4.4 3.6 7.8 8 7.8h60a8 8 0 008-8.2c-1-47.8-10.9-94.3-29.5-138.2zM512 534c-45.9 0-89.1-17.9-121.6-50.4S340 407.9 340 362c0-45.9 17.9-89.1 50.4-121.6S466.1 190 512 190s89.1 17.9 121.6 50.4S684 316.1 684 362c0 45.9-17.9 89.1-50.4 121.6S557.9 534 512 534z"
                              />
                            </svg>
                          </span>
                        </span>
                        <input
                          class="ant-input"
                          id="name"
                          placeholder="input vaccine name"
                          type="text"
                          value="FluMist"
                        />
                        <span
                          class="ant-input-suffix"
                        />
                      </span>
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="ant-form-item"
            >
              <div
                class="ant-row ant-form-item-row"
              >
                <div
                  class="ant-col ant-col-24 ant-col-offset-1 ant-form-item-label"
                >
                  <label
                    class=""
                    for="numberOfDoses"
                    title="Number of Doses"
                  >
                    Number of Doses
                  </label>
                </div>
                <div
                  class="ant-col ant-col-24 ant-form-item-control"
                >
                  <div
                    class="ant-form-item-control-input"
                  >
                    <div
                      class="ant-form-item-control-input-content"
                    >
                      <span
                        class="ant-input-affix-wrapper input ant-input-affix-wrapper-has-feedback"
                      >
                        <span
                          class="ant-input-prefix"
                        >
                          <span
                            aria-label="lock"
                            class="anticon anticon-lock"
                            role="img"
                          >
                            <svg
                              aria-hidden="true"
                              data-icon="lock"
                              fill="currentColor"
                              focusable="false"
                              height="1em"
                              viewBox="64 64 896 896"
                              width="1em"
                            >
                              <path
                                d="M832 464h-68V240c0-70.7-57.3-128-128-128H388c-70.7 0-128 57.3-128 128v224h-68c-17.7 0-32 14.3-32 32v384c0 17.7 14.3 32 32 32h640c17.7 0 32-14.3 32-32V496c0-17.7-14.3-32-32-32zM332 240c0-30.9 25.1-56 56-56h248c30.9 0 56 25.1 56 56v224H332V240zm460 600H232V536h560v304zM484 701v53c0 4.4 3.6 8 8 8h40c4.4 0 8-3.6 8-8v-53a48.01 48.01 0 10-56 0z"
                              />
                            </svg>
                          </span>
                        </span>
                        <input
                          class="ant-input"
                          id="numberOfDoses"
                          placeholder="input number of doses required"
                          type="number"
                          value="2"
                        />
                        <span
                          class="ant-input-suffix"
                        />
                      </span>
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="ant-form-item"
            >
              <div
                class="ant-row ant-form-item-row"
              >
                <div
                  class="ant-col ant-col-24 ant-col-offset-1 ant-form-item-label"
                >
                  <label
                    class=""
                    for="releaseDate"
                    title="Release Date"
                  >
                    Release Date
                  </label>
                </div>
                <div
                  class="ant-col ant-col-24 ant-form-item-control"
                >
                  <div
                    class="ant-form-item-control-input"
                  >
                    <div
                      class="ant-form-item-control-input-content"
                    >
                      <span
                        class="ant-input-affix-wrapper input ant-input-affix-wrapper-has-feedback"
                      >
                        <span
                          class="ant-input-prefix"
                        >
                          <span
                            aria-label="lock"
                            class="anticon anticon-lock"
                            role="img"
                          >
                            <svg
                              aria-hidden="true"
                              data-icon="lock"
                              fill="currentColor"
                              focusable="false"
                              height="1em"
                              viewBox="64 64 896 896"
                              width="1em"
                            >
                              <path
                                d="M832 464h-68V240c0-70.7-57.3-128-128-128H388c-70.7 0-128 57.3-128 128v224h-68c-17.7 0-32 14.3-32 32v384c0 17.7 14.3 32 32 32h640c17.7 0 32-14.3 32-32V496c0-17.7-14.3-32-32-32zM332 240c0-30.9 25.1-56 56-56h248c30.9 0 56 25.1 56 56v224H332V240zm460 600H232V536h560v304zM484 701v53c0 4.4 3.6 8 8 8h40c4.4 0 8-3.6 8-8v-53a48.01 48.01 0 10-56 0z"
                              />
                            </svg>
                          </span>
                        </span>
                        <input
                          class="ant-input"
                          id="releaseDate"
                          placeholder="input vaccine release date"
                          type="date"
                          value="2008-01-30"
                        />
                        <span
                          class="ant-input-suffix"
                        />
                      </span>
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="ant-form-item"
            >
              <div
                class="ant-row ant-form-item-row"
              >
                <div
                  class="ant-col ant-col-24 ant-col-offset-1 ant-form-item-label"
                >
                  <label
                    class=""
                    for="description"
                    title="Description"
                  >
                    Description
                  </label>
                </div>
                <div
                  class="ant-col ant-col-24 ant-form-item-control"
                >
                  <div
                    class="ant-form-item-control-input"
                  >
                    <div
                      class="ant-form-item-control-input-content"
                    >
                      <div
                        class="ant-input-textarea ant-input-textarea-in-form-item ant-input-textarea-has-feedback input input--textarea"
                        data-count="45"
                      >
                        <textarea
                          class="ant-input input input--textarea"
                          id="description"
                          rows="4"
                        >
                          Used in Seasonal Flu. Its other name is LAIV.
                        </textarea>
                        <span
                          class="ant-input-textarea-suffix"
                        />
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="form--allergy"
          >
            <div>
              <div>
                <button
                  class="ant-btn ant-btn-default button button--save"
                  type="button"
                >
                  <span>
                    Add allergy
                  </span>
                </button>
                <div />
              </div>
              <div
                class="allergy--list--container"
              />
            </div>
          </div>
        </div>
      </form>
    </div>
  </div>
</DocumentFragment>
`;
